src/main/java/de/retest/recheck/ui/diff/meta/MetadataDifference.java

```diff
@@ -23,23 +25,26 @@
 
 	private static final long serialVersionUID = 1L;
 
+	private static final MetadataDifference EMPTY = new MetadataDifference();
+
 	@Getter
 	private final Set<MetadataElementDifference> differences;
 
-	// For JAXB
 	private MetadataDifference() {
 		this( Collections.emptySet() );
 	}
 
 	public static MetadataDifference empty() {
-		return new MetadataDifference( Collections.emptySet() );
+		return EMPTY;
 	}
 
 	public static MetadataDifference of( final Set<MetadataElementDifference> differences ) {
-		final Set<MetadataElementDifference> unique =
-				new TreeSet<>( Comparator.comparing( MetadataElementDifference::getKey ) );
-		unique.addAll( differences );
-		return new MetadataDifference( new LinkedHashSet<>( unique ) );
+		return differences.isEmpty() ? empty() : differences.stream() //
+				.collect( collectingAndThen( //
+						// filter duplicate keys (keep first seen)
+						toMap( MetadataElementDifference::getKey, Function.identity(), ( first, second ) -> first ), //
+						// construct with filtered metadata
+						map -> new MetadataDifference( new HashSet<>( map.values() ) ) ) ); //
```

Comment:
I like streams and I did something similar in review, but either solution is not that readable. In theory this should never happen... . Thus I would suggest leaving this as is and discuss this separately (since the current approach has the nice benefit of the differences being sorted, if that is any argument considering).